queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py

```diff
@@ -0,0 +1,142 @@
+"""File processing rules and actions for the correction filing."""
+import datetime
+from contextlib import suppress
+from datetime import timedelta
+from typing import Dict
+
+import dpath
+from legal_api.models import Address, Business, Filing, Party, PartyRole
+
+from entity_filer.filing_meta import FilingMeta
+from entity_filer.filing_processors.filing_components import (
+    business_info,
+    create_party,
+    create_role,
+    filings,
+    update_address,
+)
+
+
+def correct_business_data(business: Business, correction_filing_rec: Filing, correction_filing: Dict,
+                          filing_meta: FilingMeta):
+    """Render the correction filing onto the business model objects."""
+    filing_meta.correction = {}
+
+    # Update business legalName if present
+    with suppress(IndexError, KeyError, TypeError):
+        name_request_json = dpath.util.get(correction_filing, '/correction/nameRequest')
+        if name_request_json.get('legalName'):
+            from_legal_name = business.legal_name
+            to_legal_name = name_request_json.get('legalName', None)
+            if to_legal_name and from_legal_name != to_legal_name:
+                business.legal_name = to_legal_name
+                filing_meta.correction = {**filing_meta.correction,
+                                          **{'fromLegalName': from_legal_name,
+                                             'toLegalName': business.legal_name}}
+    # Update Nature of Business
+    if (naics := correction_filing.get('correction', {}).get('business', {}).get('naics')) and \
+            (naics_code := naics.get('naicsCode')):
+        if business.naics_code != naics_code:
+            filing_meta.correction = {
+                **filing_meta.correction,
+                **{'fromNaicsCode': business.naics_code,
+                   'toNaicsCode': naics_code,
+                   'naicsDescription': naics.get('naicsDescription')}}
+            business_info.update_naics_info(business, naics)
+
+    # Update business office if present
+    with suppress(IndexError, KeyError, TypeError):
+        business_office_json = dpath.util.get(correction_filing, '/correction/offices/businessOffice')
+        for updated_address in business_office_json.values():
+            if updated_address.get('id', None):
+                address = Address.find_by_id(updated_address.get('id'))
+                if address:
+                    update_address(address, updated_address)
+
+    # Update parties
+    with suppress(IndexError, KeyError, TypeError):
+        party_json = dpath.util.get(correction_filing, '/correction/parties')
+        update_parties(business, party_json, correction_filing_rec)
+
+    # update court order, if any is present
+    with suppress(IndexError, KeyError, TypeError):
+        court_order_json = dpath.util.get(correction_filing, '/correction/courtOrder')
+        filings.update_filing_court_order(correction_filing_rec, court_order_json)
```

Comment:
is this required?

Reply:
Not very sure. Since we are going to use / modify this processor for any correction + we are correcting business data now, I thought it is better to have the processing handled in the filer

Reply:
ok, yeah sounds good.  Easy enough to remove after if we don't need it.